<launch>

<!-- Load controller config from the yaml file to the parameter server -->
<rosparam file="$(find rrbot_replica)/config/rrbot_control.yaml" command="load"/>

<!-- load the controllers -->
<node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/rrbot"
    args="joint_state_controller joint1_position_controller joint2_position_controller joint3_position_controller"
/>

<!-- convert joint states to tf transforms for rviz and others -->
<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    respawn="false" output="screen">

    <remap from="/joint_states" to="/rrbot/joint_states"/>

</node>


<!-- use my python script to control the arms of the robot -->

<node name="myJointControlNode" pkg="rrbot_replica" 
type="joints_control_publisher.py" output="screen" respawn="true"/>


</launch>